### ISSUE TITLE
Dynamically update menu to show dynamic items like Reader Mode

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/fenix#6275]
### Acceptance Criteria (how do I know when I’m done?)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-14128)

**Comment by Baron-Severin** [comment]
Sizing note: `BrowserMenuItem` already includes a `visible` property that can be used to determine when items are and are not shown.  It is currently only used to check visibility during `BrowserMenuAdapter` init, which occurs when the three-dot-button is clicked.  It should be possible to update this to check visibility changes more frequently, which shouldn't require any update to client code.  It will mean animating changes, and increasing complexity of the component by expanding the area of code that is forced to care about session status.  From experience, this code is already rather difficult to work with.

TL;DR:
- Will require changes to AC
- Shouldn't require any client changes
- We will need to animate these updates
  - Built in RV animations should be fine
- `browser-toolbar` is relatively difficult to work within

**Comment by NotWoods** [comment]
Related: https://github.com/mozilla-mobile/android-components/issues/5373

**Comment by csadilek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1809227

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
